test/infrastructure/docker/docker/kind_manager.go

```diff
@@ -83,89 +104,147 @@ func (m *Manager) CreateExternalLoadBalancerNode(name, image, clusterLabel, list
+		SecurityOpt: []string{"seccomp=unconfined"}, // also ignore seccomp
 		// some k8s things want to read /lib/modules
-		"--volume", "/lib/modules:/lib/modules:ro",
-		"--hostname", name, // make hostname match container name
-		"--network", defaultNetwork,
-		"--name", name, // ... and set the container name
-		// label the node with the cluster ID
-		"--label", clusterLabel,
-		// label the node with the role ID
-		"--label", fmt.Sprintf("%s=%s", nodeRoleLabelKey, role),
+		Binds:       []string{"/lib/modules:/lib/modules:ro"},
+		NetworkMode: defaultNetwork,
+		Tmpfs: map[string]string{
+			"/tmp": "", // various things depend on working /tmp
+			"/run": "", // systemd wants a writable /run
+		},
+		PortBindings: nat.PortMap{},
 	}
+	networkConfig := dockernetwork.NetworkingConfig{}
 
 	// pass proxy environment variables to be used by node's docker daemon
-	proxyDetails, err := getProxyDetails()
+	envVars := []string{}
+	proxyDetails, err := getProxyDetails(ctx)
 	if err != nil || proxyDetails == nil {
 		return nil, errors.Wrap(err, "proxy setup error")
 	}
 	for key, val := range proxyDetails.Envs {
-		runArgs = append(runArgs, "-e", fmt.Sprintf("%s=%s", key, val))
+		envVars = append(envVars, fmt.Sprintf("%s=%s", key, val))
 	}
+	containerConfig.Env = envVars
 
-	// adds node specific args
-	runArgs = append(runArgs, extraArgs...)
+	configureMounts(opts.Mounts, &hostConfig)
+	configurePortMappings(opts.PortMappings, &hostConfig)
 
-	if usernsRemap() {
+	// Expose the container port if supplied.
+	ports := nat.PortSet{}
+	if opts.Port > 0 {
+		ports[nat.Port(fmt.Sprintf("%d/tcp", opts.Port))] = struct{}{}
+	}
+
+	// We need to make sure any PortMappings are also included in the expose list
+	for port := range hostConfig.PortBindings {
+		ports[port] = struct{}{}
+	}
+	containerConfig.ExposedPorts = ports
+
+	cli, err := types.GetDockerClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "container client error")
+	}
+
+	if usernsRemap(ctx, cli) {
 		// We need this argument in order to make this command work
 		// in systems that have userns-remap enabled on the docker daemon
-		runArgs = append(runArgs, "--userns=host")
+		hostConfig.UsernsMode = "host"
 	}
 
-	if err := run(
-		image,
-		withRunArgs(runArgs...),
-		withMounts(mounts),
-		withPortMappings(portMappings),
-	); err != nil {
-		return nil, err
+	// Make sure we have the image
+	pullResp, err := cli.ImagePull(ctx, opts.Image, dockertypes.ImagePullOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "image pull error")
 	}
+	defer pullResp.Close()
 
-	return types.NewNode(name, image, role), nil
-}
+	// The ImagePull operation may take some time. We need to read the response
+	// until EOF to make sure it is complete before we move on. Output to debug
+	// level logging in case of errors.
+	pullOutput, err := io.ReadAll(pullResp)
+	if err != nil {
+		return nil, errors.Wrap(err, "image pull output error")
+	}
+	log.V(4).Info("image pull response: %s", pullOutput)
```

Comment:
We need to actually read the output to EOF. See code comment above and discussion with Stefan in the PR comments for why.

I'll bump up the verbosity level so if someone really needs it for troubleshooting they can raise the level higher. The other option I had done earlier was to just copy it out to basically `/dev/null`, but the thought was this would at least be useful if there was ever a need to debug.

Reply:
ok, let's keep this with higher verbosity